### ISSUE TITLE
Parse failure

### ISSUE THREAD

**Issue by dethe** [pocka/slack-message-parser#4]
Trying to test my `toHTML()` function, but I'm not receiving the nodes I expect. The first test I tried:

```
This is _the_ *first* ~program~ `code of the rest of this` *file*, err, _file_, and by ~file~, I mean *file*
```

Expected result:

* root
  * text
  * italic
    * text
  * text
  * bold
    * text
  * text
  * strong
    * text
  * text
  * code
    * text
  * bold
    * text
  * text
  * italic
    * text
  * text
  * strike
    * text
  * text
  * bold
    * text

Actual result:

* root
  * text
  * italic
    * text
  * text
  * bold
    * text
  * text
  * strong
    * text
  * text
  * code
    * text
  * bold
    * text

or as HTML:

expected result: 

```html
<p>This is <em>the</em> <strong>first</strong> <del>program</del> <code>code of the rest of this</code> <bold>file</bold, err, <em>file</em>, and by <del>file</del>, I mean <strong>file</strong></p>
```

actual result:

```html
<p>This is <em>the</em> <strong>first</strong> <del>program</del> <code>code of the rest of this</code> *file*, err, _file_, and by ~file~, I mean <strong>file</strong></p>

I tried another slack message parser, which failed differently. To make sure I wasn't feeding invalid slack code I tried it as a message to myself in Slack and it rendered the expected result.
```

**Comment by dethe** [comment]
Here is my entire test file, in case it helps:

```js
const {parse, NodeType} = require('slack-message-parser');
const slackdown = require('slackdown');

function wrap(tag, node){
    if (node.children){
        return wrapChildren(tag, node);
    }else if (node.text){
        return wrapText(tag, node);
    }else{
        console.warn('what am I wrapping? %o', node.type);
        return '';
    }
}

function wrapChildren(tag, node){
    return `<${tag}>${node.children.map(toHtml).join('')}</${tag}>`;
}

function wrapText(tag, node){
    return `<${tag}>${node.text}</${tag}>`;
}

function toHtml(node){
    switch(node.type){
        case NodeType.Root:
            console.log('root');
            return wrap('p', node);
        case NodeType.Text:
            console.log('text');
            return node.text;
        case NodeType.Bold:
            console.log('bold');
            return wrap('strong', node);
        case NodeType.Italic:
            console.log('italic');
            return wrap('em', node);
        case NodeType.Strike:
            console.log('strong');
            return wrap('del', node);
        case NodeType.ChannelLink:
            console.log('channel link');
            // node.channelID (string)
            // node.label (list of nodes)
            return '';
        case NodeType.UserLink:
            console.log('user link');
            // node.userID (string)
            // node.label (list of nodes)
            return '';
        case NodeType.URL:
            console.log('url');
            // node.url (string)
            // node.label (list of nodes)
            return '';
        case NodeType.Command:
            console.log('command');
            // node.name (string)
            // node.arguments (list of strings)
            // node.label (list of nodes)
            return '';
        case NodeType.Emoji:
            console.log('emoji');
            // node.name (string)
            // node.variation (string)
            return '';
        case NodeType.Code:
            console.log('code')
            return wrap('code', node)
        case NodeType.PreText:
            console.log('pre');
            return wrap('pre', node);
        case NodeType.Quote:
            console.log('quote');
            return wrap('blockquote', node);
        default:
            console.warn('Unknown type: %s', node.type);
            return '';
    }
}

function parseSlack(text){
    // const tree = parse(text);
    // return toHtml(tree);
    return slackdown.parse(text);
}
console.log(parseSlack('This is _the_ *first* ~program~ `code of the rest of this` *file*, err, _file_, and by ~file~, I mean *file*'));
```

**Comment by pocka** [comment]
Fixed in v1.0.3
